app/config.go

```diff
@@ -10,38 +10,33 @@ import (
 )
 
 type Config struct {
+	AppConfig   *models.AppConfig   `toml:"application"`
 	Connections []models.Connection `toml:"database"`
 }
 
-func LoadConfig() (config Config, err error) {
-	file, err := os.ReadFile(filepath.Join(os.Getenv("HOME"), ".config", "lazysql", "config.toml"))
-	if err != nil {
-		return
+func defaultConfig() *Config {
+	return &Config{
+		AppConfig: &models.AppConfig{
+			DefaultPageSize: 300,
+		},
 	}
-
-	err = toml.Unmarshal(file, &config)
-
-	return
 }
 
-func LoadConnections() (connections []models.Connection, err error) {
-	config, err := LoadConfig()
+func LoadConfig() error {
+	file, err := os.ReadFile(filepath.Join(os.Getenv("HOME"), ".config", "lazysql", "config.toml"))
```

Comment:
Please take a look at os.UserConfigDir

https://pkg.go.dev/os#UserConfigDir

Look at the code
https://cs.opensource.google/go/go/+/go1.23.4:src/os/file.go;l=528

The implementation is there for all OSs

Reply:
Hmm... Yeah the only issue I have with this one is that on Mac its pointing to `$HOME/Library/Application Support` instead of the more common `$HOME/.config` used for most packages you install through brew (the main packaging system for Mac). On the other hand returning `%AppData%` for Windows users is of course much, much better then `%HOME/.config`, so I'll update the code to use this instead 👍🏻

Reply:
Yeah, I know.

One solution that has been implemented many times is that you could check for the XDG_CONFIG_HOME env then falling back on os.UserConfigDir when not present